docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,162 @@ The key corresponding parts of the implementation are as follows.
+
+In the fundamental motivating use case of the HFC, `singleEraTransition` will return `Nothing` for the first ledger state of each era, and subsequent governance events within that era on the chain will cause it to return `Just` for the remaining suffix of that chain's ledger states in this era.
+However, there are a total of three possible trajectories of the `singleEraTransition` within an era on some chain.
+
+- EXTENSIBLE trajectory.
+  It starts with `Nothing` and ends with `Just`, as in the fundamental motivating use case.
+  Note that once it becomes `Just` it will always remain `Just` (with the same epoch number `e`) for the rest of this chain, by the monotonicity of `singleEraTransition` within an era of some chain.
+  This is the only case in which the above single-era interpretation of the safe zone applies, because it's the only case that involves both `Nothing` and `Just` in the same era.
+- NEVER trajectory.
+  It starts a `Nothing` and will necessarily remain `Nothing` on the entire chain.
+  For example, `singleEraTransition _cfg _params _start _st = Nothing` is a compatible definition.
+  In this case, any safe zone value would satisfy the high-level rule, because the era never ends---the implementation names this an _indefinite_ safe zone.
+- IMMEDIATE trajectory.
+  It starts as `Just e`.
+  Note that it will always remain `Just e` for the rest of this chain, by the monotonicity of `singleEraTransition` within an era of some chain.
+  The IMMEDIATE trajectory can arise in two ways.
+    - Case IMMEDIATE-INDEPENDENT.
+      It's possible that the configuration data determines the end of the era, entirely independent of the preceding chain.
+      For example, `singleEraTransition cfg _params _start _st = Just $ foo cfg + 42` is a valid definition.
+      In this case, any safe zone value would satisfy the high-level rule, because the end of the era is determined even before the start is.
+      TODO but `singleEraTransition` cannot be applied without the translate state, so what is the current code doing in this case?
+    - Case IMMEDIATE-DEPENDENT.
+      It's possible that the preceding chain before the era determines its end.
+      No known chain does this.
+      It is technically allowed by today's HFC, but only by accident.
+      There are two notable subcases; see the next subsection.
+        - Subcase IMMEDIATE-DEPENDENT-BOUND.
+          The determination instead depends only when the previous era ended.
+          This is less obviously exotic.
+          For example, the era could specify that it lasts exactly 10 epochs, regardless of when it starts.
+          In this case, any safe zone value would satisfy the high-level rule, because the end of the era is determined exactly when the start is.
+          TODO but `singleEraTransition` cannot be applied without the translate state, so what is the current code doing in this case?
```

Comment:
I explored this TODO with the `StaticTransition` type in PR https://github.com/input-output-hk/ouroboros-consensus/pull/364